pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/store/TableViewLoadDataStoreImpl.java

```diff
@@ -58,54 +59,60 @@ public TableViewLoadDataStoreImpl(PulsarClient client, String topic, Class<T> cl
     }
 
     @Override
-    public CompletableFuture<Void> pushAsync(String key, T loadData) {
+    public synchronized CompletableFuture<Void> pushAsync(String key, T loadData) {
+        if (producer == null) {
```

Comment:
nit: I think we can create `validateProducerStart` as well and reuse it.

Reply:
The return type is `CompletableFuture`; we can't throw an exception directly, right?

Reply:
Yes either way is fine. This looks good to me